### DISCUSSION TITLE
How to ingest the complex or nested column types for Trino?

### DISCUSSION THREAD

**Discussion by TheDragonWarrior** [open-metadata/OpenMetadata#5687] [Q&A]
For Trino data type, there is a type named "ROW" which can be complex structures or even nested structures, any suggestions on how to ingest these tables? Thank you!

**Comment by harshach** [comment]
@ulixius9 @pmbrull can we look into this 0.11.1 I think existing parsing should work

**Reply by pmbrull** [reply]
opened https://github.com/open-metadata/OpenMetadata/issues/5907

**Comment by pmbrull** [comment]
Hi @TheDragonWarrior, @harshach, the team gave this a go with the following table:

```
CREATE TABLE Persons1 (
         PersonID int,
         LastName varchar(255),
         FirstName varchar(255),
         Address varchar(255),
        City varchar(255),
        Test ROW(x BIGINT, y DOUBLE) );
```

And the data is getting ingested into OM:

![image](https://user-images.githubusercontent.com/35870520/177747299-ffa676dd-b8b1-4f51-bd56-638fbbb42040.png)

We also tried it out with a nested structure:

```
CREATE TABLE Persons2 (
         PersonID int,
         LastName varchar(255),
         FirstName varchar(255),
         Address varchar(255),
        City varchar(255),
        Test ROW(x BIGINT, y DOUBLE,z ROW(x1 BIGINT, y1 DOUBLE,z1 ROW(x3 BIGINT, y3 DOUBLE))) );
```

And this indeed requires some updates on our end. We'll handle this and update here once the patch is done.

Thanks, @MilanBariya for the help here!

Thanks!

**Reply by TheDragonWarrior** [reply]
Thank you @pmbrull, we will have a try on our side. Very helpful.

**Reply by pmbrull** [reply]
Update: PR being reviewed with the nested ROW fix! Thanks for reporting
